src/Cli/dotnet/Installer/Windows/RelatedProduct.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace Microsoft.DotNet.Installer.Windows
+{
+    /// <summary>
+    /// Describes a single row from the MSI <see href="https://docs.microsoft.com/en-us/windows/win32/msi/upgrade-table">Upgrade</see> 
+    /// table.
+    /// </summary>
+    internal class RelatedProduct
+    {
+        private IEnumerable<int> _languages = null;
+
+        /// <summary>
+        /// Additional flags that determine how other columns from the upgrade table are interpreted.
+        /// </summary>
+        public UpgradeAttributes Attributes
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// A comma separated list of languages identifiers (LANGID) that can be detected. When empty, all languages can be 
+        /// detected. If msidbUpgradeAttributesLanguagesExclusive is set, the list becomes exclusive.
```

Comment:
What do you mean by all languages can be detected here?

Reply:
MSIs by default store a ProductLanguage property, but in the upgrade table, you can have a list of language IDs that basically says, if you detect any of these languages being installed, you're okay to upgrade or if you have the exclusive attribute said, it means upgrade from any language except the ones listed in the Language column of the Upgrade table

![image](https://user-images.githubusercontent.com/7409981/129638663-3b7e034a-52f5-47a7-b181-bbbf95c2acef.png)

We'll likely only have 1033 in there, but I'm sticking to the installer guidelines and parsing this properly.